pkg/reconciler/taskrun/hostupdate_test.go

```diff
@@ -92,18 +101,67 @@ var _ = Describe("HostUpdateTaskRunTest", func() {
 		}
 	})
 
+	It("should fail when the config doesn't exist", func(ctx SpecContext) {
+		k8sClient := fake.NewClientBuilder().
+			WithScheme(scheme).
+			WithInterceptorFuncs(interceptor.Funcs{
+				Get: func(
+					ctx context.Context,
+					client client.WithWatch,
+					key types.NamespacedName,
+					obj client.Object,
+					opts ...client.GetOption,
+				) error {
+					return errors.NewNotFound(schema.GroupResource{
+						Group:    obj.GetObjectKind().GroupVersionKind().Group,
+						Resource: "configmaps",
+					}, obj.GetName())
+				},
+			}).Build()
+
+		log := logr.FromContextOrDiscard(ctx)
+		UpdateHostPools(testNamespace, k8sClient, &log)
+
+		list := v1.TaskRunList{}
+		Expect(k8sClient.List(ctx, &list)).To(Succeed())
+		Expect(list.Items).To(HaveLen(0))
+	})
+
 	DescribeTable("Creating taskruns for updating static hosts in a pool",
 		func(ctx SpecContext, hostConfigData map[string]string, hostSuffix string, shouldFail bool) {
 
 			hostConfig.Data = testConfigDataFromTestData(hostConfigData, hostSuffix)
 
-			k8sClient := fake.NewClientBuilder().WithScheme(scheme).WithRuntimeObjects(hostConfig).Build()
+			if !shouldFail {
```

Comment:
personally, I dislike having this kind of checks. I'd vote for splitting this test in two: failing ones, passing ones.